lib/rubocop/cop/lint/unneeded_splat_expansion.rb

```diff
@@ -124,7 +124,9 @@ def remove_brackets(array)
           elements = elements.map(&:source)
 
           if array_start.start_with?(PERCENT_W)
+            # rubocop:disable Lint/NestedDoubleQuotesInInterpolation
             "'#{elements.join("', '")}'"
```

Comment:
The distinction here is between quotes-as-code and quotes-as-content. I don't think quotes-as-content is something we should lint on. Aside from the fact that it would seem to be very difficult, there are two other problems.

* Quotes-as-content can't arbitrarily change—to make quoting clearer we can change quotes-as-code around, but we can't necessarily change content
* Quotes-as-content can be asymmetrical (apostrophes)

So while it is visually inside a single-quoted string, the quotes are content—technically it's inside a double-quoted string literal.

While the example here is annoying to read it would be less so if there was more content in the string. The disabling is necessary because it's impossible to invert the inner string literal delimiter, however now that I'm looking at it I see I could have used `%Q`.

```ruby
%Q{'#{elements.join("', '")}'}
```

Does that make sense? Do you agree?